### PR TITLE
Feature/miscellaneous improvements

### PR DESCRIPTION
**Pull request by DivineThreepwood**
### :scroll: Description

* Some minor adjustments to support the latest jul version.
* Add build dependency so that idea makes sure jul is compiled first before starting with bco.
* Improve Readme
* Deploy action inspector again as `bco-actions`
* Make sure action inspector handles port and host
* Register port and host for BCOAdhocTrainDataGenerator
* Improve exception handling.

### CODE REVIEW

File: module/app/util/src/main/java/org/openbase/bco/app/util/launch/BCOAdhocTrainDataGeneratorLauncher.java [hunk 1]
```diff
@@ -73,26 +77,31 @@
     /**
      * @param args the command line arguments
      */
-    public static void main(final String[] args) throws JPServiceException {
+    public static void main(final String[] args) throws JPServiceException, CouldNotPerformException {
         BCO.printLogo();
+        JPService.registerProperty(JPComPort.class);
+        JPService.registerProperty(JPComHost.class);
+        JPService.registerProperty(JPBCOHomeDirectory.class);
```
Comment (pLeminoq) [inline module/app/util/src/main/java/org/openbase/bco/app/util/launch/BCOAdhocTrainDataGeneratorLauncher.java:84]: What is the purpose of this property for this tool?

File: module/dal/control/src/main/java/org/openbase/bco/dal/control/layer/unit/AbstractUnitController.java [hunk 2]
```diff
@@ -1121,6 +1122,16 @@ private Action reschedule(final SchedulableAction actionToSchedule) throws Could
         }
     }
 
+    public void initRescheduling() {
+        try {
+            scheduleTimeout.restart(3, TimeUnit.MILLISECONDS);
+        } catch (CouldNotPerformException ex) {
+            if (!ExceptionProcessor.isCausedBySystemShutdown(ex)) {
+                ExceptionPrinter.printHistory("Could not rinit rescheduling!", ex, logger);
```
Comment (pLeminoq) [inline module/dal/control/src/main/java/org/openbase/bco/dal/control/layer/unit/AbstractUnitController.java]: rinit to re-init or init

**Review by pLeminoq — COMMENTED** [review]
Just a few minor comments.

File: module/dal/control/src/main/java/org/openbase/bco/dal/control/layer/unit/AbstractUnitController.java (hunk 2 above)
Comment (DivineThreepwood) [reply module/dal/control/src/main/java/org/openbase/bco/dal/control/layer/unit/AbstractUnitController.java]: fixed

**Review by pLeminoq — APPROVED** [review]

**Pull request merged**